### DISCUSSION TITLE
Unknown issue with motors leads crazyflie to crash

### DISCUSSION THREAD

**Discussion by AlexGraefe** [bitcraze/discussions#725] [Q&A] [answered]
Hi,

we currently have a problem with specific crazyflies randomly crashing during our experiments.

Background:
We use crazyswarm together with a Vicon MoCap system with 9 crazyflie flying in our testbed at the same time. We have had the problem for a while now that when only hovering, some crazyflies started to become unstable and crash to the ground. We first suspected a wrong setup of the MoCap-System and crazyswarm caused this problem. We tested different marker arrangements (both unique/non-unique), and different numbers and positions of cameras. Currently the mocap sees all markers on the drones, even when the drone starts to become unstable, without any ghost markers and crazyswarm also gives no error/warning messages. We also freshly calibrate the system every day after warming it up. We also tried different channels in crazyradio to exclude interference on the 2.4 GHz band.  So we are fairly confident that the MoCap system and crazyswarm do not cause this problem.
The problem is also independent of the firmware. It occurs with our custom firmware and the vanilla firmware of the crazyflie. It is also independent of our additional custom deck. The drone crashes with and without it.

The interesting thing is that when I swap the drone that is crashing with another drone, flash the same firmware on the drone and use the same battery, this new drone does not crash. We first suspected the IMU of the crazyflie to be damaged and somehow disturb the estimation of the state. However, after we swaped all 4 motors and propeller on this crazyflie, it flew stable again.

For some drones it is even sometimes sufficient to unplug and replug the motors, although there is no visible loose contact between motor and plug.

I have multiple ideas what is the cause of this issue:

1. In our experiments, the drones tend to fall to the ground from time to time (from 1m-3m) due to software-bugs. Although we have mats on the ground and nets around the testbed, this might have damaged some motors. However, the first drone we bought 2.5 years ago and crashed the most still flies as new and the issue tends to occur more often with drones that we bought 1 month ago.
2. There is a quality issue in one batch of the motors causing them to malfunction.
3. Although the connector between motor and crazyflie-PCB looks good, it has a loose contact and replugging it fixes this temporarily.

Is this issue known and do you have an idea what causes this issue/how to avoid it?

Thank you for your answer.

devjalx

**Comment by krichardsson** [comment, marked as answer]
Hi,
that is an extensive investigation! Good job!

Tobias, our hardware expert is enjoying a vacation this week, but I'll try to add what I know for now, and hopefully Tobias has some more input next week.

A possible explanation to the crashing of your Crazyflies might be related to that the motors do not have enough "umpf" to enable the controller to handle some situations. We see a clear difference in this type of "random" crashes when using the motor upgrade kit that has a bit more thrust. https://store.bitcraze.io/products/thrust-upgrade-bundle-for-crazyflie-2-x?_pos=1&_sid=ac28f6350&_ss=r 
Maybe one reason for this could be that the current to the motors is being limited?

I have not heard about oxidation or other dirt on the motor connectors, but on the other hand your investigation is very thorough and I don't think that many other users would check them out. One possibility could be that there is a bit of flux from the manufacturing process that has not been cleaned out properly, but this is not something we have seen. You can try to clean the connectors with some isopropanol to see if it makes a difference.

A similar problem that we do have seen, is that the battery connector may get some oxidation over time, which increases resistance. You might want to clean this as well.

**Reply by AlexGraefe** [reply]
Hi,

thank you for your answer. 

> Maybe one reason for this could be that the current to the motors is being limited?

I thought there is no current control for the motors. Is there a parameter that limits the current to the motors?

We cleaned the contacts of battery and motor and have not seen a significant difference. However, we noticed that maybe the motor holders might also be a possible reason. Some of them seemed to be a bit loose/easy to twist, maybe because of crashes in the past. 

Because we use a lot of drones, the probability of at least one drone not working because of one small fault is high. So maybe it is also a combination of multiple different faults.

The additional deck and the markers add a weight of 10g to the crazyflie. Maybe this also favors instabilities in combination with issues with the motor holders/contacts. 
We ordered the stronger motors/propellers. They should arrive in one week and we will give you feedback how they worked.

**Reply by tobbeanton** [reply]
I can also add that the brushed coreless motors don't have that great of a lifetime until the brushes wear out. Perhaps around 30 hours during normal usage. It does not seem to be the problem in your case though.

**Reply by AlexGraefe** [reply]
Hi,

the upgrade kits have finally arrived. It seems that the drones fly more stable. We will see in the long run if this improves the robustness.

To summarize, if crazyflies are used frequently and have sudden unstable behavior, swapping the motors might solve the problem.

Thank you very much for your help.

**Comment by krichardsson** [comment]
> I thought there is no current control for the motors. Is there a parameter that limits the current to the motors?

You are correct, what I meant was that oxidation on the battery connector for instance, adds a resistance which will limit the current available to the motors.

Like you say, all small problems adds up and reduces the probability of a successful flight. With multiple drones the probability for the full swarm gets worse quickly with increasing numbers of individuals.

Looking forward to hear if the situation is improved with the upgrade kit!
